File: app/backend/src/controllers/authController.ts
import { Request, Response } from 'express';
import { users } from '../data/users';
import { generateToken } from '../utils/jwt';

export const login = (req: Request, res: Response) => {
  const { username, password } = req.body;

  if (!username || !password) {
    return res.status(400).json({
      success: false,
      message: 'Username and password required'
    });
  }

  const user = users.find(
    u => u.username === username &&
         u.password === password
  );

  if (!user) {
    return res.status(401).json({
      success: false,
      message: 'Invalid credentials'
    });
  }
 
 const token = generateToken({
  id: user.id,
  username: user.username,
  role: user.role
});

  return res.json({
  success: true,
  user: {
    id: user.id,
    username: user.username,
    role: user.role
  },
  token
});
};

File: app/backend/FOMS.API/Controllers/AuthController.cs
using FOMS.API.Models;
using FOMS.API.Services;
using Microsoft.AspNetCore.Mvc;

namespace FOMS.API.Controllers;

[ApiController]
[Route("api/auth")]
public class AuthController : ControllerBase
{
    private readonly AuthService _authService;

    public AuthController(AuthService authService)
    {
        _authService = authService;
    }

    [HttpPost("login")]
    public IActionResult Login(LoginRequest request)
    {
        if (string.IsNullOrWhiteSpace(request.Username) ||
            string.IsNullOrWhiteSpace(request.Password))
        {
            return BadRequest(new
            {
                success = false,
                message = "Username and password required"
            });
        }

        var user = _authService.Authenticate(request);

        if (user == null)
        {
            return Unauthorized(new
            {
                success = false,
                message = "Invalid credentials"
            });
        }

        return Ok(new
        {
            success = true,
            user
        });
    }
}